onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1731,7 +1731,7 @@ Status GemmOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const N
                         "bias of QDQGemm should be int32, actual type: ", bias_tensor.data_type());
       Shape bias_dimen;
       for (auto dim : bias_tensor.dims())
-        bias_dimen.push_back(dim);
+        bias_dimen.push_back(SafeInt<uint32_t>(dim));
```

Comment:
oh for this line here, I was suggested that the SafeInt here is not required as it's only needed when doing calculations that lead to memory allocations in ORT code, so removed it before. but not sure.

Reply:
ah, this change was to address a compiler warning about the different data types. I just copied this approach from above:
https://github.com/microsoft/onnxruntime/blob/ae043e3963eabf16522b0b1ef23531acc6a20977/onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc#L1694